### DISCUSSION TITLE
Is it possible to filter products by nested custom model fields ?

### DISCUSSION THREAD

**Discussion by guillaumeklaus** [swellstores/community#124] [Help] [answered]
Hello ! 

I'm starting to work with Swell, thanks for the amazing job!
I have a question about filtering, let suppose that I have 2 custom models : 

```
- Event: 
name: string
 openingDate: datetime 
 closingDate: datetime 
```
```
 - SubEvent: 
name: string
event: Event
```

And a product linked to a `SubEvent`.

Is it possible to filter all products to keep only those that are before or after an `openingDate` in the nested model `Event`?
Something like : `curl https://api.swell.store/products?where[sub_event.event.openingDate][$gte]=2018-01-01T00:00:00Z`

I tried both with curl query and with the Node sdk, but none seems to work.

Thanks!

**Comment by ericingram** [comment, marked as answer]
Are those models nested arrays, or collections? Could you post the structure from `/:models/products`?

**Reply by guillaumeklaus** [reply]
The link between `Product` and `SubEvent` seems to be a `Lookup` one and it's the same for the link between `Event` and `SubEvent`.

Here is the structure (a little summarized) of /:models/products : 

```
{
  "version": "1.0.114",
  "name": "products",
  "api": "com",
  "label": "Product",
  "plural": "Products",
  "fields": {
  ...,
      "sub_event_id": {
      "type": "objectid",
      "content_id": "custom.admin.product-fields"
    },
    "sub_event": {
      "type": "link",
      "content_id": "custom.admin.product-fields",
      "model": "sub_event",
      "key": "sub_event_id"
    }},
   ...
  } 
```

**Reply by ericingram** [reply]
It's not yet possible to filter records through model links. You'd have to find the matching `events`, then fetch related products from there.

**Reply by guillaumeklaus** [reply]
Ok! I will do it this way

Thanks for your quick reply 🙏
